bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/TemplateUtils.java

```diff
@@ -37,4 +47,94 @@ public static CloudBigtableTableConfiguration BuildImportConfig(ImportOptions op
+   */
+  private static class RequestWithScanValueProvider
+      implements ValueProvider<ReadRowsRequest>, Serializable {
+    private final ValueProvider<String> start;
+    private final ValueProvider<String> stop;
+    private final ValueProvider<Integer> maxVersion;
+    private final ValueProvider<String> filter;
+    private ReadRowsRequest cachedRequest;
+
+    RequestWithScanValueProvider(
+        ValueProvider<String> start,
+        ValueProvider<String> stop,
+        ValueProvider<Integer> maxVersion,
+        ValueProvider<String> filter) {
+      this.start = start;
+      this.stop = stop;
+      this.maxVersion = maxVersion;
+      this.filter = filter;
+    }
+
+    @Override
+    public ReadRowsRequest get() {
+      if (cachedRequest == null) {
+        Scan scan = new Scan();
+        if (start.get() != null && !start.get().isEmpty()) {
+          scan.setStartRow(stop.get().getBytes());
+        }
+        if (stop.get() != null && !stop.get().isEmpty()) {
+          scan.setStopRow(stop.get().getBytes());
+        }
+        if (maxVersion.get() != null) {
+          scan.setMaxVersions(maxVersion.get());
+        }
+        if (filter.get() != null && !filter.get().isEmpty()) {
+          try {
+            scan.setFilter(new ParseFilter().parseFilterString(filter.get()));
+          } catch (CharacterCodingException e) {
+            e.printStackTrace();
+            throw new RuntimeException(e);
+          }
+        }
+
+        ReadHooks readHooks = new DefaultReadHooks();
+        ReadRowsRequest.Builder builder = Adapters.SCAN_ADAPTER.adapt(scan, readHooks);
+        cachedRequest = readHooks.applyPreSendHook(builder.build());
+      }
+      return cachedRequest;
+    }
+
+    @Override
+    public boolean isAccessible() {
+      return start.isAccessible()
+          && stop.isAccessible()
+          && maxVersion.isAccessible()
+          && filter.isAccessible();
+    }
+
+    @Override
+    public String toString() {
+      if (isAccessible()) {
+        return String.valueOf(get());
+      }
+      return CloudBigtableConfiguration.VALUE_UNAVAILABLE;
+    }
+  }
+
+  /** Builds CloudBigtableScanConfiguration from input runtime parameters for export job. */
+  public static CloudBigtableScanConfiguration BuildExportConfig(ExportOptions opts) {
+    ValueProvider<ReadRowsRequest> request =
+        new RequestWithScanValueProvider(
+            opts.getBigtableStartRow(),
+            opts.getBigtableStopRow(),
+            opts.getBigtableMaxVersions(),
+            opts.getBigtableFilter());
+    CloudBigtableScanConfiguration.Builder configBuilder =
+        new CloudBigtableScanConfiguration.Builder()
+            .withProjectId(opts.getBigtableProject())
+            .withInstanceId(opts.getBigtableInstanceId())
+            .withTableId(opts.getBigtableTableId())
+            .withRequest(request);
+
+    if (opts.getBigtableAppProfileId() != null) {
```

Comment:
app profile id is a value provider, the null check doesn't make sense

Reply:
Yeap, I was going to chain them up and without checking null, I will add withAppProfileId()
 to CloudBigtableTableConfiguration.Builder and CloudBigtableScanConfiguration.Builder first.

Reply:
updated/merged.